org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/AnalyzeProjectJob.java

```diff
@@ -474,13 +439,37 @@ public Map<String, String> getAnalysisProperties() {
 
   }
 
-  public AnalysisResults run(final AnalysisConfiguration analysisConfig, final Map<ISonarLintIssuable, List<RawIssue>> issuesPerResource, IProgressMonitor monitor) {
-    SonarLintLogger.get().debug("Starting analysis with configuration:\n" + analysisConfig.toString());
-    var issueListener = new SonarLintIssueListener(getProject(), issuesPerResource);
-    var result = runAnalysis(analysisConfig, issueListener, monitor);
-    SonarLintLogger.get().info("Found " + issueListener.getIssueCount() + " issue(s)");
-    return result;
+  private AnalyzeFilesResponse run(final Map<ISonarLintFile, IDocument> docPerFiles, final Map<String, String> extraProps,
+    final Map<ISonarLintIssuable, List<RawIssueDto>> issuesPerResource, long startTime, IProgressMonitor monitor) {
+    var analysisId = UUID.randomUUID();
+    var analysisState = new AnalysisState(analysisId, getProject(), issuesPerResource);
+    try {
+      RunningAnalysesTracker.get().track(analysisState);
+      var future = SonarLintBackendService.get().analyzeFiles(getProject(), analysisId, docPerFiles, extraProps, startTime);
+      AnalyzeFilesResponse response = waitForFuture(monitor, future);
+      SonarLintLogger.get().info("Found " + analysisState.getIssueCount() + " issue(s)");
+      return response;
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new CanceledException();
+    } catch (ExecutionException e) {
+      throw new IllegalStateException(e);
+    } finally {
+      RunningAnalysesTracker.get().finish(analysisState);
+    }
   }
 
-  protected abstract AnalysisResults runAnalysis(AnalysisConfiguration analysisConfig, SonarLintIssueListener issueListener, IProgressMonitor monitor);
+  private static <T> T waitForFuture(IProgressMonitor monitor, CompletableFuture<T> future) throws InterruptedException, ExecutionException {
```

Comment:
This implementation is not needed, we already have a [JobUtils.waitForFuture](https://github.com/SonarSource/sonarlint-eclipse/blob/10.0.1.81733/org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/utils/JobUtils.java#L53-L64), please re-use this one.

Reply:
The `JobUtils.waitForFuture` behavior is a bit different than the method I created here, because I wanted to explicitly throw a `CancelationException` in the context of a `Job`. But I moved the method to `JobUtils` so it can be reused (I renamed it to `waitForFutureInJob`), and at some point we might need to harmonize how we handle those cases in SLE.